src/components/common/mediaPicker/MediaPickerSidebarContainer.js

```diff
@@ -0,0 +1,95 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { injectIntl } from 'react-intl';
+import { connect } from 'react-redux';
+import { initializePreviouslySelectedMedia, clearSelectedMedia, resetMetadataShortlist } from '../../../actions/systemActions';
+
+function composeMediaPickerSidebarContainer() {
+  return (ChildComponent) => {
+    class MediaPickerSidebarContainer extends React.Component {
+      state = {
+        // queryInStore: false,
+      };
+
+      UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
Can we use something not unsafe here?

Reply:
I don't know of the right way to do this? I'd rather keep the pattern until we modify all the files that use this

Reply:
Hooks is new in React 16.8 and I guess useState() would be the answer to this, but I don't think we want to go this route

Reply:
I'm not sure that hooks are the solution here (or maybe there are multiple ones). From https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html:

> getDerivedStateFromProps and getSnapshotBeforeUpdate.

appear to be the replacements.